### ISSUE TITLE
Avoid refereshing project model multiple times in parallel

### ISSUE THREAD

**Issue by mkaput** [starkware-libs/cairo#6237]
Avoid calling `scarb metadata` if another instance is already running (but if one is running: always schedule at most 1 extra run to pick any changes that could happen while the previous run was running).

**Comment by mkaput** [comment]
actually this is duplicate of #6128
